application/src/main/java/org/togetherjava/tjbot/commands/moderation/KickCommand.java

```diff
@@ -0,0 +1,108 @@
+package org.togetherjava.tjbot.commands.moderation;
+
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.Interaction;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+import java.util.Objects;
+
+
+/**
+ * This command can kicks users. Kicking can also be paired with a kick reason. The command will
+ * also try to DM the user to inform them about the action and the reason.
+ * <p>
+ * The command fails if the user triggering it is lacking permissions to either kick other users or
+ * to kick the specific given user (for example a moderator attempting to kick an admin).
+ *
+ */
+public final class KickCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(KickCommand.class);
+    private static final String TARGET_OPTION = "user";
+    private static final String REASON_OPTION = "reason";
+
+    /**
+     * Constructs an instance.
+     */
```

Comment:
```suggestion
```

Reply:
Public members must have javadoc. Its unfortunate that we can not configure it to exclude no-arg constructors or similar.